src/libsyntax/parse/parser.rs

```diff
@@ -5809,13 +5811,19 @@ impl<'a> Parser<'a> {
                                     maybe_append(attrs, extra_attrs));
             return Ok(Some(item));
         }
-        if self.check_keyword(keywords::Unsafe) &&
-            self.look_ahead(1, |t| t.is_keyword(keywords::Impl))
+        if (self.check_keyword(keywords::Unsafe) &&
```

Comment:
I'm adding a test taken from [src/test/run-pass/specialization/defaultimpl/specialization-basics.rs](https://github.com/giannicic/rust/blob/defaultimpl/src/test/run-pass/specialization/defaultimpl/specialization-basics.rs) putting `unsafe` to all occurences of `default impl`.
    Probably it's incorrect to look for `default unsafe impl`, I'm changing in order to check for `unsafe default impl`.